addons/mail/static/src/models/composer/composer.js

```diff
@@ -118,6 +120,31 @@ registerModel({
                 textInputSelectionDirection: clear(),
             });
         },
+        /**
+         * Try to fetch link preview when urls are inserted inside the composer.
+         */
+        getLinkPreview(pastedText) {
+            const urls = pastedText.match(urlRegexp) || false;
+            if (!urls) {
+                return;
+            }
+            // Add attachment for urls that are not already in attachment
+            const existingUrls = this.attachments.map(att => att.url);
+            const filteredUrls = urls.filter(url => !existingUrls.includes(url));
+            for (const url of filteredUrls) {
+                this.performRPClinkPreview(url);
+            }
+        },
+        async performRPClinkPreview(url) {
+            const attachmentData = await this.env.services.rpc({
+                route: '/mail/link_preview',
+                params: { url, channel_id: this.activeThread.id },
+            }, { shadow: true });
+            if (!attachmentData) {
+                return;
+            }
+            this.update({ attachments: insert(attachmentData) });
```

Comment:
Would be best to just `this.messaging.models['Attachment'].insert(attachmentData)`.
The relation between `Composer` and `Attachment` should be automatic from `attachmentData` (and computed fields).

Reply:
Your solution isn't working. Probably because attachments are not in fact computed for composer.

Reply:
Ok, was kinda expecting it.
This shows modelling is not declarative enough.
It's not urgent to fix in this PR, but just be aware of this problem at least.